spec/javascripts/modules/cross-domain-tracking.spec.js

```diff
@@ -0,0 +1,97 @@
+describe('Cross Domain Tracking', function () {
+  'use strict'
+  var module
+
+  beforeEach(function () {
+    GOVUK.analytics = GOVUK.analytics || {}
+    GOVUK.analytics.addLinkedTrackerDomain = function () {
+    }
+
+    spyOn(GOVUK.analytics, 'addLinkedTrackerDomain')
+    module = new GOVUK.Modules.CrossDomainTracking()
+  })
+
+  it('tracks realistic example', function () {
+    var anchorToTest = document.createElement('a')
+    anchorToTest.href = 'https://www.registertovote.service.gov.uk/register-to-vote/start'
+    anchorToTest.className = 'button button--start'
+    anchorToTest.setAttribute('role', 'button')
+    anchorToTest.setAttribute('data-module', 'cross-domain-tracking')
+    anchorToTest.setAttribute('data-tracking-code', 'UA-23066786-5')
+    anchorToTest.setAttribute('data-tracking-name', 'transactionTracker')
+    anchorToTest.textContent = 'Start Now'
+
+    module.start($(anchorToTest))
+
+    expect(
+      GOVUK.analytics.addLinkedTrackerDomain
+    ).toHaveBeenCalledWith('UA-23066786-5', 'transactionTracker', 'www.registertovote.service.gov.uk')
+  })
+
+  it('tracks links with cross-domain-analytics data attributes', function () {
+    var anchorToTest = document.createElement('a')
+    anchorToTest.href = 'https://www.gov.uk/browse/citizenship/voting'
+    anchorToTest.setAttribute('data-module', 'cross-domain-tracking')
+    anchorToTest.setAttribute('data-tracking-code', 'UA-XXXXXXXXX-Y')
+    anchorToTest.setAttribute('data-tracking-name', 'govspeakButtonTracker')
+
+    var wrapperDiv = document.createElement('div')
+    wrapperDiv.appendChild(anchorToTest)
+
+    module.start($(wrapperDiv))
+
+    expect(
+      GOVUK.analytics.addLinkedTrackerDomain
+    ).toHaveBeenCalledWith('UA-XXXXXXXXX-Y', 'govspeakButtonTracker', 'www.gov.uk')
+  })
+
+  it('tracks multiple links', function () {
+    var anchorToTest = document.createElement('a')
+    anchorToTest.href = 'https://www.gov.uk/browse/citizenship/voting'
+    anchorToTest.setAttribute(
+      'data-tracking-code',
+      'UA-XXXXXXXXX-Y'
+    )
+    anchorToTest.setAttribute(
+      'data-tracking-name',
+      'govspeakButtonTracker'
+    )
+
+    var secondAnchorToTest = document.createElement('a')
+    secondAnchorToTest.href = 'https://www.registertovote.service.gov.uk/register-to-vote/start'
+    secondAnchorToTest.setAttribute(
+      'data-tracking-code',
+      'UA-23066786-5'
+    )
+    secondAnchorToTest.setAttribute(
+      'data-tracking-name',
+      'transactionTracker'
+    )
+
+    var wrapperDiv = document.createElement('div')
+    wrapperDiv.appendChild(anchorToTest)
+    wrapperDiv.appendChild(secondAnchorToTest)
```

Comment:
Again, some markup like the following would make it clearer what the test is doing and the expected markup needed for the module to work:

```
<div>
  <a href="#"></a>
  <a href="#"></a>
</div>
```

Reply:
Like: https://github.com/alphagov/static/blob/master/spec/javascripts/modules/track-click.spec.js#L140-L156

Reply:
Same as above